tests/unit/test_filter_tree.cpp

```diff
@@ -399,16 +419,26 @@ TEST_F(FilterTreeTest, FilterTree_Compact) {
 	ASSERT_TRUE(FilterTree_Compact(actual));
 	compareFilterTrees(expected, actual);
 	FilterTree_Free(actual);
+	FilterTree_Free(expected);
 
+	// Non compactable 'true' tree.
 	actual = _build_deep_non_compactable_or_true_tree();
```

Comment:
How about each of these functions, e.g. `_build_deep_non_compactable_or_true_tree` will return both `actual` but also `expected` ?
it is a bit strange or out of context to see a query here...
`FT_FilterNode *_build_deep_non_compactable_or_true_tree(FT_FilterNode **expected)`

Reply:
done